app/src/js/components/Rules/add.js

```diff
@@ -1,99 +1,158 @@
+ * @param {string} b - second string to compare
+ * @param {number} a number less than 0 if `a` is "less than" `b`, 0 if they
+ *    are "equal" (ignoring case), or a number greater than 0 if `a` is
+ *    "greater than" `b`
+ */
+const asc = (a, b) => a.localeCompare(b, 'en', { sensitivity: 'base' });
 
-const defaultRuleObject = {
+const initialRule = {
   name: '',
   workflow: '',
   provider: '',
-  collection: {
-    name: '',
-    version: ''
-  },
+  collection: '',
   meta: {},
   rule: {
     type: '',
     value: ''
   },
-  state: 'ENABLED'
+  state: ''
 };
 
-const AddRule = ({ rules, location = {}, dispatch, schema, ...rest }) => {
-  const [defaultValue, setDefaultValue] = useState(defaultRuleObject);
+const AddRule = ({
+  dispatch,
+  rules,
+  collections,
+  providers,
+  workflows,
+  location = {}
+}) => {
+  const [rule, setRule] = useState(initialRule);
   const { state: locationState } = location;
   const { name } = locationState || {};
-  const { rule: ruleSchema } = schema || {};
   const { map: rulesMap } = rules || {};
-  const isCopy = !!name;
+  const isCopy = name !== undefined;
   const title = isCopy ? 'Copy a rule' : 'Add a rule';
+  const [enums, setEnums] = useState({
+    collection: [],
+    provider: [],
+    workflow: []
+  });
 
   useEffect(() => {
-    if (isCopy) {
-      dispatch(getSchema('rule'));
+    const { data } = rulesMap[name] || {};
+
+    if (isCopy && data) {
+      // Set the rule to the entry in the rules map associated with the rule's
+      // name, but first replace the collection name/version object with the
+      // collection's ID (string) since the "Add/Copy Rule" form displays the
+      // colleciton ID's in a dropdown.
+      setRule({ ...data, 'collection': getCollectionId(data.collection) });
     }
-  }, [isCopy, dispatch]);
+  }, [name, rulesMap, isCopy]);
 
   useEffect(() => {
-    const record = rulesMap[name];
-    const { data } = record || {};
-    if (isCopy && data && ruleSchema) {
-      setDefaultValue(removeReadOnly(data, ruleSchema));
-    }
-  }, [ruleSchema, name, rulesMap, isCopy]);
+    if (!collections.list.meta.queriedAt) dispatch(listCollections());
+    if (!providers.list.meta.queriedAt) dispatch(listProviders());
+    if (!workflows.list.meta.queriedAt) dispatch(listWorkflows());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dispatch]);
+
+  useEffect(() => {
+    setEnums({
+      collection: collections.list.data.map(getCollectionId).sort(asc),
+      provider: providers.list.data.map(({ id }) => id).sort(asc),
+      workflow: workflows.list.data.map(({ name }) => name).sort(asc)
+    });
+  }, [collections, providers, workflows]);
 
   return (
-    <AddRaw
-      pk={'new-rule'}
-      title={title}
+    <AddRecord
+      data={rule}
+      schemaKey={'rule'}
       primaryProperty={'name'}
+      title={title}
       state={rules}
-      defaultValue={defaultValue}
+      baseRoute={'/rules/rule'}
```

Comment:
Somethings up here, because clicking Cancel brings you back to `/rules/rules`, which is a blank page